### PR TITLE
Upgrade terraform-provider-github to v5.2.0

### PR DESCRIPTION
**Pull request by lblackstone**
Breaking changes:
- Branches attribute moved from Repository resource to a new RepositoryBranches data source

Fix https://github.com/pulumi/pulumi-github/issues/242

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 PACK := github
 ORG := pulumi
 PROJECT := github.com/$(ORG)/pulumi-$(PACK)
-PROVIDER_PATH := provider/v4
```
Comment (danielrbradley) [inline Makefile:6]: Have you relected this change in ci-mgmt yet?

**Review by danielrbradley — APPROVED** [review]
Looks fine on the surface without going into digging into details. Any specifcs you'd want more sense-checking?

File: Makefile (hunk 1 above)
Comment (lblackstone) [reply Makefile:6]: No, I forgot that was a thing. I'll check on that!

**Comment by lblackstone** [comment]
> Looks fine on the surface without going into digging into details. Any specifcs you'd want more sense-checking?

I haven't done many bridged provider upgrades, let alone for a major version bump, so I don't know what to suggest. Anyone who knows more, feel free to chime in with corrections!

File: Makefile (hunk 1 above)
Comment (lblackstone) [reply Makefile:6]: https://github.com/pulumi/ci-mgmt/pull/260

**Comment by guineveresaenger** [comment]
You'll want to add `VERSION_PREFIX=5.0.0` to `.github/master.yml` so the Go SDK builds properly; this will get overridden once the related ci-mgmt changes drop.

Also if there's any go sdk acceptance tests, skip them here and re-add them in a follow-up PR. 

Docs here: https://github.com/pulumi/platform-providers-team/blob/main/playbooks/tf-provider-major-version-update.md

**Review by guineveresaenger — APPROVED** [review]
LGTM; thanks!

**Pull request merged**